### ISSUE TITLE
Is it possible to use the dot "." button on the keyboard instead of the comma one?

### ISSUE THREAD

**Issue by frederikhors** [Canutin/svelte-currency-input#32]
Thank you very much! This is absolutely amazing!

I would like to use the "." dot button on my keyboard instead of the comma one.

Is it possible?

**Comment by fmaclen** [comment]
@frederikhors somehow I was expecting to get notified of issues in this repo but didn't, my apologies.

Not sure what you mean exactly, are you asking if you could use `.` to separate digits and `,` as the decimal point?

**Comment by frederikhors** [comment]
Nope. If I need to write 43,56 dollars or euro I would like to use the dot for decimal points too other than comma.

**Comment by fmaclen** [comment]
What determines which decimal point character you can use is the locale.

For example:
- Locale `en-US` — Currency `EUR` —> `€43.56`
- Locale `de-DE` — Currency `EUR` —> `43,56 €`

The idea (from a UX point of view) is that the field should use the decimal point that a "normal user" would be used to given the set locale.

What the input doesn't do is (at least at the moment) is to allow you to press `.` and convert it to a `,` if a locale like `de-DE` is set.

**Comment by frederikhors** [comment]
> What the input doesn't do is (at least at the moment) is to allow you to press `.` and convert it to a `,` if a locale like `de-DE` is set.

This is what I'm asking. Because many people use it and it's annoying to choose comma just to adhere to local conventions.

**Comment by fmaclen** [comment]
I hear ya. If you are more of a _computer person_ I can see how you are used to using "." for decimals regardless of the locale.

From an implementation perspective it should be straight forward, what I'd need to test a bit is how that behavior works if you are user that don't expect to use "." for decimals.

If it gets too in the way maybe I can add an option to the API that would be disabled by default but you could enable in your specific case.

**Comment by frederikhors** [comment]
I think we can convert the dot in comma (and the inverse) so no issue for the user.

**Comment by fmaclen** [comment]
@frederikhors thanks for the suggestion, works much better than I was expecting.
[Take it for a drive](https://www.npmjs.com/package/@canutin/svelte-currency-input/v/0.7.0) and let me know if you find anything weird.

**Comment by frederikhors** [comment]
It doesn't work, as you can see below in the GIF.

I'm trying to write € 18523,46 using the dot.

![iss-currency](https://user-images.githubusercontent.com/41120635/192260305-61b9e8c9-412a-4955-a305-9cb21f6f71b8.gif)

**Comment by fmaclen** [comment]
Oh, I see what the issue is.

**Comment by fmaclen** [comment]
@frederikhors give it another shot.

**Comment by frederikhors** [comment]
It works! Thanks!
